src/aiida/engine/processes/functions.py

```diff
@@ -435,7 +439,7 @@ def define(cls, spec):
                     def indirect_default(value=default):
                         return to_aiida_type(value)
                 else:
-                    indirect_default = default
+                    indirect_default = default  # type: ignore[assignment]
```

Comment:
Not super clear how to solve this one, since `indirect_default` can be basically anything

> src/aiida/engine/processes/functions.py:438: error: Incompatible types in assignment (expression has type "Any | tuple[()]", variable has type "Callable[[Any], Any]")  [assignment]

Reply:
If it can be anything then as error message suggest use `Callable[[Any], Any]` ?

Reply:
I am not sure what are you suggesting here? the callable type is inferred because `indirect_default` is defined as a function in the if branch above, but in the else branch in can be anything. I think this we could squash this by inverting the logic and put the else branch first, but I don't really want to touch the logic.

Reply:
Okay, I misunderstood the change here.

I think the mypy complain because the typechecker resolve the expression in if..else.. branch and want to make sure the type of expression conform with each other. 

Then the correct way to solve this might be for the else branch do:

```python
def indirect_default(value=default):
    return lambda _: value
```

(I am not sure this is correct, but I think it is better to just ignore the type error here to not touch the logic. so if you don't want to try please ignore my comment.)